irods/pool.py

```diff
@@ -73,6 +81,7 @@ def get_connection(self):
                     # code more predictable as we are not relying on when garbage collector is called
                     conn.disconnect()
                     conn = Connection(self, self.account)
+                    new_conn = True
                     logger.debug("Created new connection with id: {}".format(id(conn)))
             except KeyError:
                 conn = Connection(self, self.account)
```

Comment:
Do we need to set `new_conn` to `True` after this line as well?

Reply:
Yes indeed, good call.